### PR TITLE
Make seek work on `/dev/null`

### PR DESCRIPTION
**Pull request by hoodmane**
Seeking `/dev/null` should work and do nothing instead of returning `EPIPE`. This fixes it.

### CODE REVIEW

File: test/other/test_seek_dev_null.c [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    if (fd == -1) {
+        printf("open failed: %s\n", strerror(errno));
+        return 1;
+    }
+    int res = lseek(fd, 10, SEEK_CUR);
+    printf("res: %d\n", res);
+    if (res == -1) {
+        printf("lseek failed: %s\n", strerror(errno));
+        return 1;
+    }
```
Comment (sbc100) [inline test/other/test_seek_dev_null.c:19]: It looks like our existing /dev/null testing is in `test/test_files.c`.. perhaps we can just add this to that existing test?

**Review by sbc100 — APPROVED** [review]
LGTM % comment

**Comment by hoodmane** [comment]
Okay I moved the test to `test_files.c` as requested.

**Comment by hoodmane** [comment]
Is the CI stuck here?

**Comment by sbc100** [comment]
Hmm.. can you try rebasing or merging, to see if it will trigger?

**Review by sbc100 — APPROVED** [review]

**Pull request merged**